mantidproject/mantid
# S(Q, w) table checks

**Description of work.**
https://docs.mantidproject.org/nightly/algorithms/SofQWNormalisedPolygon-v1.html#algm-sofqwnormalisedpolygon now has validateInputs function to check that the table is of the correct form (int, double, double), that the input workspace is of the correct type and that the table and workspace have the same number of detectors. 

**To test:**

The following script should create an invalid table and should be caught at the validate inputs stage. If we run the algorithm from the GUI then it would give the red star next to the table.
```
import numpy as np

#Create Table to hold detector angular range
IRIS_PG_Det = CreateEmptyTableWorkspace()

#Calculate range for PG detectors on IRIS
no_det = int(51)
det_start = int(3)
detIDList = [i+det_start for i in range(no_det)]
angle_start = np.radians(25.)
angle_end = np.radians(160.)
angles = np.linspace(angle_start, angle_end, no_det+1, endpoint = True)
min = angles[:-1]
max = angles[1:]

#Add some columns, Recognized types are: int,float,double,bool,str,V3D,long64
IRIS_PG_Det.addColumn(type="int",name="Detector ID")
IRIS_PG_Det.addColumn(type="int",name="Min two theta")
IRIS_PG_Det.addColumn(type="double",name="Max two theta")

#Populate the columns for three detectors
for i,j in enumerate(detIDList):
    IRIS_PG_Det.addRow ([j, i, max[i] ] )
```

Fixes #26637 . 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
